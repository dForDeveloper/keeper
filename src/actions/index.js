export const setNotes = (notes) => ({
  type: 'SET_NOTES',
  notes
});

export const addNote = (note) => ({
  type: 'ADD_NOTE',
  note
});

export const updateNote = (note) => ({
  type: 'UPDATE_NOTE',
  note
});

export const deleteNote = (id) => ({
  type: 'DELETE_NOTE',
  id
});

export const toggleLoading = (bool) => ({
  type: 'TOGGLE_LOADING',
  bool
});

export const setError = (message) => ({
  type: 'SET_ERROR',
  message
});

export const setStatus = (code) => ({
  type: 'SET_STATUS',
  code
});

export const setQuery = (query) => ({
  type: 'SET_QUERY',
  query
});

export const setUser = (user) => ({
  type: 'SET_USER',
  user
});