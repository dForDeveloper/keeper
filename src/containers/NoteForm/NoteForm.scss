.NoteForm {
  z-index: 1;
  position: fixed;
  background-color: #fff;
  width: 50%;
  left: 25%;
  top: 13%;
  border-radius: 10px;
  border: solid 2px #004ecb;
}

.NoteForm--title,
.NoteForm--list-item,
.NoteForm--new-input,
.NoteForm--p--complete {
  margin: 15px auto;
  display: inline-block;
  width: 80%;
  border: none;
  outline: none;
  text-align: left;
  border-bottom: 1px solid black;
}

.NoteForm--p--complete {
  text-decoration: line-through;
  color: lightgray;
}

.NoteForm--title {
  font-size: 1.6rem;
}

.NoteForm--span--incomplete,
.NoteForm--span--complete,
.NoteForm--new-input {
  display: block;
}

.NoteForm--icon--unchecked,
.NoteForm--icon--checked,
.NoteForm--icon--delete {
  width: 15px;
}