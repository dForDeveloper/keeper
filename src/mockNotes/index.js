export const mockNotes = [
  { 
    id: 'ijf', 
    title: 'Learn Redux', 
    listItems: [
      { 
        id: 'asg', 
        description: 'read through lesson', 
        isComplete: false
      }, 
      { 
        id: 'rhf', 
        description: 'follow along with tutorial', 
        isComplete: false
      }
    ]
  },
  { 
    id: 'pdo', 
    title: 'Start portfolio', 
    listItems: [
      { 
        id: 'cat', 
        description: 'Setup user account', 
        isComplete: false
      }
    ]
  },
  { 
    id: 'bah', 
    title: 'Grocery shopping', 
    listItems: [
      { 
        id: 'rey', 
        description: 'Stop by King Soopers', 
        isComplete: false
      }, 
      { 
        id: 'moo', 
        description: 'Stop by Sprouts', 
        isComplete: false
      }
    ]
  },
  { 
    id: 'fah', 
    title: 'Make dinner', 
    listItems: [
      { 
        id: 'ley', 
        description: 'gather ingredients', 
        isComplete: false
      }, 
      { 
        id: 'mei', 
        description: 'preheat oven', 
        isComplete: false
      }
    ]
  }
]

export const mockNote = {
  id: 'rzz',
  title: 'Workout',
  listItems: [
    {
      id: 'lpo',
      description: '20 situps',
      isComplete: false
    },
    {
      id: 'lqw',
      description: '15 pushups',
      isComplete: false
    }
  ]
}