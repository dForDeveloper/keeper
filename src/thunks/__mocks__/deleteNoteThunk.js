export const deleteNoteThunk = jest.fn();