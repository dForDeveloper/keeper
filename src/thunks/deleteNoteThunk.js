import { fetchData, createOptions } from '../utils/api';
import { setError, deleteNote, toggleLoading, setStatus } from '../actions';

export const deleteNoteThunk = (id) => {
  return async (dispatch) => {
    const url = `http://localhost:3001/api/v1/notes/${id}`;
    const options = createOptions('DELETE');
    try {
      dispatch(toggleLoading(true));
      const response = await fetchData(url, options);
      dispatch(toggleLoading(false));
      dispatch(setStatus(response.status));
      dispatch(deleteNote(id));
    } catch (error) {
      dispatch(setError(error.message));
    }
  }
}