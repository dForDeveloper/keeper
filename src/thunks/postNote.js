import { fetchData, createOptions } from '../utils/api';
import { addNote, setError, toggleLoading, setStatus } from '../actions';

export const postNote = (note) => {
  return async (dispatch) => {
    const { title, listItems } = note;
    const url = 'http://localhost:3001/api/v1/notes/';
    const options = createOptions('POST', { title, listItems });
    try {
      dispatch(toggleLoading(true));
      const response = await fetchData(url, options);
      const note = await response.json();
      dispatch(toggleLoading(false));
      dispatch(addNote(note));
      dispatch(setStatus(response.status));
    } catch (error) {
      dispatch(setError(error.message));
    }
  }
}