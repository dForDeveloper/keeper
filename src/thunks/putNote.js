import { fetchData, createOptions } from '../utils/api';
import { setError, updateNote, toggleLoading, setStatus } from '../actions';

export const putNote = (note) => {
  const { id, title, listItems } = note;
  return async (dispatch) => {
    const url = `http://localhost:3001/api/v1/notes/${id}`;
    const options = createOptions('PUT', { title, listItems });
    try {
      dispatch(toggleLoading(true));
      const response = await fetchData(url, options);
      dispatch(toggleLoading(false));
      dispatch(updateNote(note));
      dispatch(setStatus(response.status));
    } catch (error) {
      dispatch(setError(error.message));
    }
  }
}