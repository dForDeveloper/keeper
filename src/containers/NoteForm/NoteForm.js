import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Redirect, Link } from 'react-router-dom';
import shortid from 'shortid';
import CompleteItem from '../../components/CompleteItem/CompleteItem';
import IncompleteItem from '../../components/IncompleteItem/IncompleteItem';
import { putNote } from '../../thunks/putNote';
import { postNote } from '../../thunks/postNote';
import { deleteNoteThunk } from '../../thunks/deleteNoteThunk';
import { setStatus } from '../../actions';
import paletteicon from '../../images/paletteicon.svg';

export class NoteForm extends Component {
  constructor() {
    super();
    this.state = {
      title: '',
      listItems: [],
      focusedListItemID: null,
      color: 'white',
      showColorOptions: false
    }
  }

  componentDidMount() {
    const { path } = this.props.match;
    const { title, listItems, color } = this.props;
    if (path !== '/new-note') {
      this.setState({ title, listItems, color });
    }
  }

  createListItem = (id, description) => ({
    id,
    description,
    isComplete: false
  })

  editListItems = (listItems, id, description) => {
    return listItems.map(item => {
      return item.id === id ? { ...item, description } : item;
    });
  }

  getIncompleteListItems = () => {
    const { focusedListItemID, listItems } = this.state;
    const incompleteItems = listItems.filter(item => !item.isComplete);
    return incompleteItems.map(item => {
      return (
        <IncompleteItem
          key={item.id}
          id={item.id}
          description={item.description}
          focusedListItemID={focusedListItemID}
          handleComplete={this.handleComplete}
          handleChange={this.handleChange}
          handleItemDelete={this.handleItemDelete}
        />
      );
    });
  }

  getCompleteListItems = () => {
    const { listItems } = this.state;
    const completeItems = listItems.filter(item => item.isComplete);
    return completeItems.map(item => {
      return (
        <CompleteItem
          key={item.id}
          id={item.id}
          description={item.description}
          handleComplete={this.handleComplete}
          handleItemDelete={this.handleItemDelete}
        />
      );
    });
  }

  getNewListItemInput = () => (
    <input
      name={shortid.generate()}
      value=''
      onChange={this.handleChange}
      placeholder='List item'
      className='NoteForm--new-input'
    />
  )

  getTitleInput = () => (
    <input
      name='title'
      value={this.state.title}
      placeholder='Title'
      onChange={(event) => this.setState({ title: event.target.value })}
      className='NoteForm--title'
    />
  )

  handleChange = (event) => {
    const { name: id, value: description } = event.target;
    const { listItems } = this.state;
    const existingListItem = listItems.find(item => item.id === id);
    let updatedListItems;
    if (existingListItem) {
      updatedListItems = this.editListItems(listItems, id, description);
    } else {
      updatedListItems = [...listItems, this.createListItem(id, description)];
    }
    this.setState({ listItems: updatedListItems, focusedListItemID: id });
  }

  handleColorChoice = (event) => {
    const color = event.target.id;
    this.setState({ color });
  }

  handleComplete = (id) => {
    const { listItems } = this.state;
    const updatedListItems = listItems.map(item => {
      const { isComplete } = item;
      return item.id === id ? { ...item, isComplete: !isComplete } : item;
    });
    this.setState({ listItems: updatedListItems });
  }

  handleItemDelete = (id) => {
    const { listItems } = this.state;
    const updatedListItems = listItems.filter(item => item.id !== id);
    this.setState({ listItems: updatedListItems });
  }

  handleNoteDelete = async () => {
    const { id } = this.props.match.params;
    await this.props.deleteNoteThunk(id);
    this.props.setStatus(0);
  }

  handleSubmit = async (event) => {
    event.preventDefault();
    const { id } = this.props.match.params;
    const { title, listItems, color } = this.state;
    if (id) {
      await this.props.putNote({ id, title, listItems, color });
    } else {
      await this.props.postNote({ title, listItems, color });
    }
    this.props.setStatus(0);
  }

  toggleShowColorOptions = () => {
    const { showColorOptions } = this.state;
    this.setState({ showColorOptions: !showColorOptions });
  }

  render() {
    const { title, color, showColorOptions } = this.state;
    const { status } = this.props;
    const { path } = this.props.match;
    return (
      <form className={'NoteForm--background-' + color} onSubmit={this.handleSubmit}>
        {this.getTitleInput()}
        {this.getIncompleteListItems()}
        {this.getNewListItemInput()}
        {this.getCompleteListItems()}
        <button
          className={'NoteForm--button-' + color}
          disabled={title.trim() === ''}
          // onClick={this.handleSubmit}
        >
          Save
        </button>
        {path !== '/new-note' ?
          <button className={'NoteForm--button-' + color} onClick={this.handleNoteDelete}>
            Delete
          </button> :
          <Link to='/'>
            <button className={'NoteForm--button-' + color}>Discard</button>
          </Link>}
        {showColorOptions &&
          <div className='NoteForm--color-options'>
            <button id='white' className='NoteForm--color' onClick={this.handleColorChoice}></button>
            <button id='pink' className='NoteForm--color' onClick={this.handleColorChoice}></button>
            <button id='orange' className='NoteForm--color' onClick={this.handleColorChoice}></button>
            <button id='yellow' className='NoteForm--color' onClick={this.handleColorChoice}></button>
            <button id='green' className='NoteForm--color' onClick={this.handleColorChoice}></button>
            <button id='blue' className='NoteForm--color' onClick={this.handleColorChoice}></button>
            <button id='lavender' className='NoteForm--color' onClick={this.handleColorChoice}></button>
            <button id='gray' className='NoteForm--color' onClick={this.handleColorChoice}></button>
          </div>
        }
        <div className='NoteForm--div--palette-icon'>
          <img className='NoteForm--palette-icon' src={paletteicon} alt='palette icon' onClick={this.toggleShowColorOptions} />
        </div>
        {(status >= 200 && status < 300) && <Redirect to='/' />}
      </form>
    )
  }
}

export const mapStateToProps = (state) => ({
  status: state.status
});

export const mapDispatchToProps = (dispatch) => ({
  putNote: (note) => dispatch(putNote(note)),
  postNote: (note) => dispatch(postNote(note)),
  deleteNoteThunk: (id) => dispatch(deleteNoteThunk(id)),
  setStatus: (code) => dispatch(setStatus(code))
});

export default connect(mapStateToProps, mapDispatchToProps)(NoteForm);

NoteForm.propTypes = {
  deleteNoteThunk: PropTypes.func,
  putNote: PropTypes.func,
  postNote: PropTypes.func,
  status: PropTypes.number,
  setStatus: PropTypes.func
}