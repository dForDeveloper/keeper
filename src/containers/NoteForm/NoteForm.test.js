import React from 'react';
import { shallow } from 'enzyme';
import { NoteForm, mapStateToProps, mapDispatchToProps } from './NoteForm';
import { mockNote, mockUpdatedNote, mockNotes } from '../../mockNotes';
import { deleteNoteThunk } from '../../thunks/deleteNoteThunk';
import { putNote } from '../../thunks/putNote';
import { postNote } from '../../thunks/postNote';
import { setStatus } from '../../actions';

jest.mock('shortid', () => ({ generate: jest.fn(() => '') }));
jest.mock('../../thunks/deleteNoteThunk.js');
jest.mock('../../thunks/putNote.js');
jest.mock('../../thunks/postNote.js');

describe('NoteForm', () => {
  let wrapper;
  let wrapperNewNote;
  const { id, title, listItems } = mockNote;
  const mockMatch = { params: { id }, path: `/notes/${id}` }
  const mockMatchNewNote = { params: {}, path: '/new-note' }
  const mockProps = {
    match: mockMatch,
    title,
    listItems,
    status: 0,
    deleteNoteThunk: jest.fn(() => true),
    putNote: jest.fn(() => true),
    postNote: jest.fn(() => true),
    setStatus: jest.fn(() => true)
  }
  const mockPropsNewNote = {...mockProps, match: mockMatchNewNote}
  const newListItem = {
    id: 'newid',
    description: 'new description',
    isComplete: false
  }
  const updatedListItem = {
    id: 'lpo',
    description: 'eat',
    isComplete: false
  }
  const completedListItem = {
    id: 'lpo',
    description: '20 situps',
    isComplete: true
  }
  
  beforeEach(() => {
    wrapper = shallow(<NoteForm {...mockProps}/>);
    wrapperNewNote = shallow(<NoteForm {...mockPropsNewNote}/>);
  });

  it('should have default state', () => {
    expect(wrapperNewNote.state()).toEqual({
      title: '',
      listItems: [],
      focusedListItemID: null
    });
  });

  it('should match the snapshot when the path is /new-note', () => {
    expect(wrapperNewNote).toMatchSnapshot();
  });

  it('should match the snapshot when the path is /notes/:id', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot when 200 < status < 300', () => {
    wrapper = shallow(<NoteForm {...mockProps} status={200} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('componentDidMount', () => {
    it('should set state when path is not /new-note', () => {
      expect(wrapper.state('title')).toEqual(title);
      expect(wrapper.state('listItems')).toEqual(listItems);
    });
  });

  describe('createListItem', () => {
    it('should return a listItem', () =>{
      const { id, description } = newListItem;
      const result = wrapper.instance().createListItem(id, description);
      expect(result).toEqual(newListItem);
    });
  });

  describe('editListItem', () => {
    it('should return an updated array of listItems', () => {
      const expected = [updatedListItem, listItems[1]];
      const result = wrapper.instance().editListItems(listItems, 'lpo', 'eat');
      expect(result).toEqual(expected);
    });
  });

  describe('getListItems', () => {
    it('should return an array of IncompleteItems and CompleteItems', () => {
      const { listItems } = mockUpdatedNote;
      wrapper.setState({ listItems });
      const incomplete = '.NoteForm--span--incomplete';
      const complete = '.NoteForm--span--complete';
      const result = wrapper.instance().getListItems();
      const wrapperIndex0 = shallow(result[0]);
      const wrapperIndex1 = shallow(result[1]);
      const wrapperIndex2 = shallow(result[2]);
      expect(wrapperIndex0.find(incomplete)).toHaveLength(1);
      expect(wrapperIndex1.find(incomplete)).toHaveLength(1);
      expect(wrapperIndex2.find(complete)).toHaveLength(1);
    });
  });

  describe('getNewListItemInput', () => {
    it('should return an new listItem input element', () => {
      const result = wrapper.instance().getNewListItemInput();
      const wrapperInput = shallow(result);
      const inputClass = '.NoteForm--new-input';
      expect(wrapperInput.find(inputClass)).toHaveLength(1);
    });
  });

  describe('getTitleInput', () => {
    it('should return a title input element with the correct value', () => {
      const result = wrapper.instance().getTitleInput();
      const wrapperInput = shallow(result);
      const inputClass = '.NoteForm--title';
      expect(wrapperInput.find(inputClass)).toHaveLength(1);
      expect(wrapperInput.find('[value="Workout"]')).toHaveLength(1);
    });

    it('should set state if the title input changes', () => {
      const mockEvent = { target: { value: 'A' } };
      wrapperNewNote.find('.NoteForm--title').simulate('change', mockEvent);
      expect(wrapperNewNote.state('title')).toEqual('A');
    });
  });

  describe('handleChange', () => {
    it('should set state if the event target is an existing list item', () => {
      const { id, description } = updatedListItem;
      const mockEvent = { target: { name: id, value: description } }
      const expected = [updatedListItem, listItems[1]];
      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state('focusedListItemID')).toEqual(id);
      expect(wrapper.state('listItems')).toEqual(expected);
    });

    it('should set state if the event target is a new list item', () => {
      const { id, description } = newListItem;
      const mockEvent = { target: { name: id, value: description } }
      const expected = [...listItems, newListItem];
      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state('focusedListItemID')).toEqual(id);
      expect(wrapper.state('listItems')).toEqual(expected);
    });
  });

  describe('handleComplete', () => {
    it('should set state with updated list items', () => {
      const expected = [completedListItem, listItems[1]];
      wrapper.instance().handleComplete('lpo');
      expect(wrapper.state('listItems')).toEqual(expected);
    });
  });

  describe('handleItemDelete', () => {
    it('should set state with remaining list items', () => {
      const expected = [listItems[1]];
      wrapper.instance().handleItemDelete('lpo');
      expect(wrapper.state('listItems')).toEqual(expected);
    });
  });

  describe('handleNoteDelete', () => {
    it('should call deleteNoteThunk with the correct param', async () => {
      wrapper.instance().handleNoteDelete();
      expect(mockProps.deleteNoteThunk).toHaveBeenCalledWith(id);
    });

    it('should call setStatus with the correct param', () => {
      wrapper.instance().handleNoteDelete();
      expect(mockProps.setStatus).toHaveBeenCalledWith(0);
    });
  });

  describe('handleSubmit', () => {
    it('should call putNote if there is an id', () => {
      wrapper.instance().handleSubmit();
      expect(mockProps.putNote).toHaveBeenCalledWith(mockNote);
    });

    it('should call postNote if there is not an id', () => {
      const expected = { title: 'new note', listItems: [] };
      wrapperNewNote.instance().setState(expected);
      wrapperNewNote.instance().handleSubmit();
      expect(mockProps.postNote).toHaveBeenCalledWith(expected);
    });

    it('should call setStatus with the correct param', () => {
      wrapper.instance().handleSubmit();
      expect(mockProps.setStatus).toHaveBeenCalledWith(0);
    });
  });

  describe('mapStateToProps', () => {
    it('should return an object with a status property', () => {
      const mockState = {
          notes: mockNotes,
          isLoading: false,
          error: '',
          status: 0
      };
      const expected = { status: 0 };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    const mockDispatch = jest.fn();

    it('should call dispatch with a setStatus action', () => {
      const actionToDispatch = setStatus(204);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setStatus(204);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('should call dispatch with a deleteNoteThunk thunk', () => {
      const actionToDispatch = deleteNoteThunk('lpo');
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.deleteNoteThunk('lpo');
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('should call dispatch with a postNote thunk', () => {
      const actionToDispatch = postNote(mockNote);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.postNote(mockNote);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('should call dispatch with a putNote thunk', () => {
      const actionToDispatch = putNote(mockNote);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.putNote(mockNote);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});