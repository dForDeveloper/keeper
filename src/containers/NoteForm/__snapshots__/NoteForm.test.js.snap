// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteForm should match the snapshot when 200 < status < 300 1`] = `
<form
  className="NoteForm--background-lavender"
  onSubmit={[Function]}
>
  <input
    className="NoteForm--title"
    name="title"
    onChange={[Function]}
    placeholder="Title"
    value="Workout"
  />
  <IncompleteItem
    description="20 situps"
    focusedListItemID={null}
    handleChange={[Function]}
    handleComplete={[Function]}
    handleItemDelete={[Function]}
    id="lpo"
    key="lpo"
  />
  <input
    className="NoteForm--new-input"
    name=""
    onChange={[Function]}
    placeholder="List item"
    value=""
  />
  <CompleteItem
    description="15 pushups"
    handleComplete={[Function]}
    handleItemDelete={[Function]}
    id="lqw"
    key="lqw"
  />
  <button
    className="NoteForm--button-lavender"
    disabled={false}
  >
    Save
  </button>
  <button
    className="NoteForm--button-lavender"
    onClick={[Function]}
  >
    Delete
  </button>
  <div
    className="NoteForm--div--palette-icon"
  >
    <img
      alt="palette icon"
      className="NoteForm--palette-icon"
      onClick={[Function]}
      src="paletteicon.svg"
    />
  </div>
  <Redirect
    push={false}
    to="/"
  />
</form>
`;

exports[`NoteForm should match the snapshot when the path is /new-note 1`] = `
<form
  className="NoteForm--background-white"
  onSubmit={[Function]}
>
  <input
    className="NoteForm--title"
    name="title"
    onChange={[Function]}
    placeholder="Title"
    value=""
  />
  <input
    className="NoteForm--new-input"
    name=""
    onChange={[Function]}
    placeholder="List item"
    value=""
  />
  <button
    className="NoteForm--button-white"
    disabled={true}
  >
    Save
  </button>
  <Link
    replace={false}
    to="/"
  >
    <button
      className="NoteForm--button-white"
    >
      Discard
    </button>
  </Link>
  <div
    className="NoteForm--div--palette-icon"
  >
    <img
      alt="palette icon"
      className="NoteForm--palette-icon"
      onClick={[Function]}
      src="paletteicon.svg"
    />
  </div>
</form>
`;

exports[`NoteForm should match the snapshot when the path is /notes/:id 1`] = `
<form
  className="NoteForm--background-lavender"
  onSubmit={[Function]}
>
  <input
    className="NoteForm--title"
    name="title"
    onChange={[Function]}
    placeholder="Title"
    value="Workout"
  />
  <IncompleteItem
    description="20 situps"
    focusedListItemID={null}
    handleChange={[Function]}
    handleComplete={[Function]}
    handleItemDelete={[Function]}
    id="lpo"
    key="lpo"
  />
  <input
    className="NoteForm--new-input"
    name=""
    onChange={[Function]}
    placeholder="List item"
    value=""
  />
  <CompleteItem
    description="15 pushups"
    handleComplete={[Function]}
    handleItemDelete={[Function]}
    id="lqw"
    key="lqw"
  />
  <button
    className="NoteForm--button-lavender"
    disabled={false}
  >
    Save
  </button>
  <button
    className="NoteForm--button-lavender"
    onClick={[Function]}
  >
    Delete
  </button>
  <div
    className="NoteForm--div--palette-icon"
  >
    <img
      alt="palette icon"
      className="NoteForm--palette-icon"
      onClick={[Function]}
      src="paletteicon.svg"
    />
  </div>
</form>
`;
